File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        if (nums.length > 1) {
            int i = 0;
            int count = 0;
            while (i < nums.length) {
                if (nums[i] != 0) {
                    nums[count] = nums[i];
                    count++;
                }
                i++;
            }
            while (count < nums.length) {nums[count++] = 0;}
        }
    }
}

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        l,r = 0,1
        while(r < len(nums)):
            if nums[l] == 0:
                while(r < len(nums) and nums[r] == 0):
                    r += 1
                if r < len(nums):
                    nums[l], nums[r] = nums[r], nums[l]
                l, r = l+1, r+1
            else:
                l, r = l+1, r+1


        